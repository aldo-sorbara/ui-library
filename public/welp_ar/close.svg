<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
