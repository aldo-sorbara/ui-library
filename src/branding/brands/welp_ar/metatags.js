export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
