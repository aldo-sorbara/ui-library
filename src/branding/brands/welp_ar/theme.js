import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
