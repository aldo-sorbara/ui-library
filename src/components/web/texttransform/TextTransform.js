import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

/**
 * @render react
 * @name TextTransform
 * @description TextTransform Component for Wenance
 */

class TextTransform extends React.Component {
  componentDidMount() {
    const { modifiers } = this.props;
    let { text } = this.props;
    modifiers.map(m => {
      text = (this.modifiers[m] || this.modifiers.def)(text);
      return null;
    });
    this.refs.component && (this.refs.component.innerHTML = text);
  }

  safeMatch = (str, regexp) => str.match(regexp) || [];

  createRegExp = (preRegx, postRegx = preRegx) =>
    new RegExp(`${preRegx}(.*?)${postRegx}`, 'g');

  modifiers = {
    def: t => t,
    bold: str =>
      this.safeMatch(str, this.createRegExp('\\*')).reduce(
        (toReturn, s) =>
          (toReturn = toReturn.replace(
            s,
            `<b>${s.substring(1, s.length - 1)}</b>`
          )),
        str
      ),
    italic: str =>
      this.safeMatch(str, this.createRegExp('\\_')).reduce(
        (toReturn, s) =>
          (toReturn = toReturn.replace(
            s,
            `<i>${s.substring(1, s.length - 1)}</i>`
          )),
        str
      ),
    links: str =>
      this.safeMatch(str, this.createRegExp('(\\[', '\\]){2}')).reduce(
        (toReturn, s) => {
          const [text, link] = this.safeMatch(
            s,
            this.createRegExp('\\[', '\\]')
          );
          return toReturn.replace(link, '').replace(
            text,
            `<a href="${link.substring(1, link.length - 1)}" target="_blank">
	  				  ${text.substring(1, text.length - 1)}
	  			  </a>`
          );
        },
        str
      )
  };

  render() {
    const Component = StyledTextTransform;
    return <Component {...this.props} ref={'component'} />;
  }
}

const StyledTextTransform = styled.span`
  margin: 0;
`;

TextTransform.defaultProps = {
  modifiers: []
};

TextTransform.propTypes = {
  modifiers: PropTypes.array,
  text: PropTypes.string.isRequired
};

export default TextTransform;
