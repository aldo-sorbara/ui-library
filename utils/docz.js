export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
