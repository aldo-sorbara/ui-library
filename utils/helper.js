import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
