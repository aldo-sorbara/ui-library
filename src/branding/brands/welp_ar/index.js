import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
