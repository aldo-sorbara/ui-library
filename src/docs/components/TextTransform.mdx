---
name: TextTransform
menu: Components
route: /texttransform
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import TextTransform from '../../components/web/texttransform';
import Input from '../../components/web/input';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import { Description } from '../../components/web/global';
import GlobalStyle from '../../components/web/globalstyle';
import Checkbox from '../../components/web/checkbox';
import { styles } from '../../../utils/docz';

# TextTransform

<PropsTable of={TextTransform} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <TextTransform modifiers={['bold']} text={'this *text* must be *bold*'} />
      <TextTransform
        modifiers={['italic']}
        text={'this _text_ must be _italic_'}
      />
      <TextTransform
        modifiers={['italic', 'bold']}
        text={'this text must be _italic_ and this *bold*'}
      />
      <TextTransform
        modifiers={['links']}
        text={
          'Acepto los [Términos y Condiciones][https://welponlinecomar.com/wp-content/uploads/2019/01/welp-tyc.pdf] and [another link][http://localhost:3001]'
        }
      />
      <TextTransform modifiers={['links']} text='non special text' />
      <Description>
        <TextTransform
          modifiers={['italic', 'bold']}
          text={'this text must be _italic_ and this *bold*'}
        />
      </Description>
      <Checkbox>
        <TextTransform
          element='p'
          modifiers={['links']}
          text={
            'Acepto los [¿Términos y Condiciones?][https://welponlinecomar.com/wp-content/uploads/2019/01/welp-tyc.pdf] and [another link][http://localhost:3001]'
          }
        />
      </Checkbox>
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/texttransform--with-default-props)
