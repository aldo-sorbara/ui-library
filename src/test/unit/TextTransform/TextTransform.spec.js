import React from 'react';
import TextTransform from '../../../components/web/texttransform';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('TextTransform component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <TextTransform
          modifiers={['links']}
          text={
            'Acepto los [Términos y Condiciones][https://welponlinecomar.com/wp-content/uploads/2019/01/welp-tyc.pdf] and [another link][http://localhost:3001]'
          }
        />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render text with a tag', () => {
    const wrapper = mountWithTheme(
      <TextTransform
        modifiers={['links']}
        text={
          'Acepto los [Términos y Condiciones][https://welponlinecomar.com/wp-content/uploads/2019/01/welp-tyc.pdf] and [another link][http://localhost:3001]'
        }
      />,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toContain('links');
    expect(typeof wrapper.props().text).toBe('string');
    expect(wrapper.render().find('a').length).toBe(2);
  });

  it('should render text with b tag', () => {
    const wrapper = mountWithTheme(
      <TextTransform
        modifiers={['bold']}
        text={'this *text* must be *bold*'}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toContain('bold');
    expect(typeof wrapper.props().text).toBe('string');
    expect(wrapper.render().find('a').length).toBe(0);
    expect(wrapper.render().find('b').length).toBe(2);
  });

  it('should render text with i tag', () => {
    const wrapper = mountWithTheme(
      <TextTransform
        modifiers={['italic']}
        text={'this _text_ must be _italic_'}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toContain('italic');
    expect(typeof wrapper.props().text).toBe('string');
    expect(wrapper.render().find('b').length).toBe(0);
    expect(wrapper.render().find('i').length).toBe(2);
  });

  it('should render text', () => {
    const wrapper = mountWithTheme(
      <TextTransform
        modifiers={['default']}
        text={'this _text_ must be _italic_'}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toContain('default');
    expect(typeof wrapper.props().text).toBe('string');
    expect(wrapper.render().find('b').length).toBe(0);
  });

  it('should render non special text', () => {
    const wrapper = mountWithTheme(
      <TextTransform modifiers={['bold']} text={'non special text'} />,
      welp_ar.theme
    );
    expect(wrapper.props().modifiers).toContain('bold');
    expect(typeof wrapper.props().text).toBe('string');
    expect(wrapper.render().find('b').length).toBe(0);
  });
});
