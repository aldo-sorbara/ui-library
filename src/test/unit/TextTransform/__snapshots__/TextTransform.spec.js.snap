// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextTransform component should render snapshot 1`] = `
.c0 {
  margin: 0;
}

<span
  className="c0"
/>
`;
